bia_explorer/api.py

```diff
@@ -0,0 +1,392 @@
+        ]
+
+    def get_study(self) -> BIAStudy:
+        return ApiClient.get_study_by_uuid(self.study_uuid)
+
+    def repr_html_embed_image(self) -> Optional[str]:
+        example_image_annotation = [
+            annotation
+            for annotation in self.annotations
+            if annotation.key == 'example_image_uri'
+        ]
+        if not len(example_image_annotation):
+            return
+        
+        example_image_annotation = example_image_annotation.pop()
+        return example_image_annotation.value
+
+
+class ImageSlice(BaseModel):
+    c: Optional[int] = None
+    x: Optional[int] = None
+    y: Optional[int] = None
+    z: Optional[int] = None
+    t: Optional[int] = None
+
+    def as_tuple(self):
+        # ?!
+        return (self.c, self.z, self.t)
+        
+class BIAImageRepresentation(api_models.BIAImageRepresentation, ReprHtmlMixin):
+    """
+    ! No representation-image link
+    """
+    def ome_ngff_to_dask_array(self):
+        zgroup = zarr.open(self.uri[0], mode='r')
+        zarray = zgroup['0']
+
+        return da.from_zarr(zarray)
+
+    def to_ome_zarr(self):
+        reader = Reader(parse_url(self.uri[0]))
+        
+        return reader
+
+    def zipped_zarr_to_dask_array(self):
+        raise Exception("TODO")
+
+    def to_bytesio(self, max_size_bytes = 100000000) -> BytesIO:
+        # Always check size first
+        file_size = requests.get(self.uri[0], stream=True).headers['Content-length']
+        file_size = int(file_size)
+        if max_size_bytes and file_size > max_size_bytes:
+            raise Exception(f"File size {file_size} over the maximum limit of {max_size_bytes}. Pass a higher max_size_bytes (or None to disable) if your machine has enough memory")
+
+        data_request = requests.get(self.uri[0])
+        assert data_request.status_code == 200
+
+        img = BytesIO(data_request.content)
+        return Image.open(img)
+
+    def to_dask_array(self):
+        if self.type == ImageRepresentationType.OME_NGFF.value:
+            return self.ome_ngff_to_dask_array()
+        elif self.type == ImageRepresentationType.OME_ZARR_ZIPPED.value:
+            return self.zipped_zarr_to_dask_array()
+        else:
+            raise Exception(f"No automatic display supported for representation of type {self.type} at '{self.uri[0]}'")
+
+    def pyplot_display_slice(self, image_slice: Optional[ImageSlice] = None):
+        if image_slice is None:
+            # @TODO: Make "default slice" some function of the actual darray 
+            image_slice = ImageSlice(c=0,t=0,z=0)
+
+        slice_tuple = image_slice.as_tuple()
+        img_as_darray = self.to_dask_array()
+        slice_to_display = img_as_darray[slice_tuple].compute()
+        if len(np.shape(slice_to_display)) != 2:
+            raise Exception(f"Unable to display slice of shape {np.shape(slice_to_display)}. xy plane required")
+
+        plt.imshow(slice_to_display, cmap='gray')
```

Comment:
I'd say return a Pillow Image or something 

Image.fromarray(slice_to_display)

The reason being is the pilot plays nicely with notebooks and non notebooks and matplotlib (plt.show()) doesn't always.

And or, just return the numpy array and let the users use it in the notebooks (manually)

Reply:
In general I prefer the Pillow image, but one thing matplotlib does is scaling, pure Image.fromarray will do painful things if the image plane is 40000x50000. This might actually want to refuse to show if the plane is too big.